src/twisted/internet/test/test_posixbase.py

```diff
@@ -291,10 +315,18 @@ def stopListening():
         self.assertTrue(self.called)
 
 
-class WakerTests(TestCase):
+class WakerTests(WarningCheckerTestCase):
     def test_noWakerConstructionWarnings(self):
+        """
+        No warnings are generated when constructing the waker.
```

Comment:
this is a test failing on Windows.
it was failing as it checked there are no warnings.

But there were warnings that were accumulated by running other tests.

I re-wrote a bit the test while trying to understand what it does.